pype/plugins/global/publish/extract_review.py

```diff
@@ -13,21 +16,1085 @@ class ExtractReview(pyblish.api.InstancePlugin):
     families = ["review"]
     hosts = ["nuke", "maya", "shell", "premiere"]
 
+    # Supported extensions
+    image_exts = ["exr", "jpg", "jpeg", "png", "dpx"]
+    video_exts = ["mov", "mp4"]
+    supported_exts = image_exts + video_exts
+
+    # FFmpeg tools paths
+    ffmpeg_path = pype.lib.get_ffmpeg_tool_path("ffmpeg")
+
+    # Preset attributes
+    profiles = None
+
+    # Legacy attributes
     outputs = {}
     ext_filter = []
     to_width = 1920
     to_height = 1080
 
     def process(self, instance):
+        # ffmpeg doesn't support multipart exrs
+        if instance.data.get("multipartExr") is True:
+            instance_label = (
+                getattr(instance, "label", None)
+                or instance.data.get("label")
+                or instance.data.get("name")
+            )
+            self.log.info((
+                "Instance \"{}\" contain \"multipartExr\". Skipped."
+            ).format(instance_label))
+            return
+
+        # Use legacy processing when `profiles` is not set.
+        if self.profiles is None:
+            return self.legacy_process(instance)
+
+        # Run processing
+        self.main_process(instance)
+
+        # Make sure cleanup happens and pop representations with "delete" tag.
+        for repre in tuple(instance.data["representations"]):
+            tags = repre.get("tags") or []
+            if "delete" in tags:
+                instance.data["representations"].remove(repre)
+
+    def main_process(self, instance):
+        host_name = pyblish.api.registered_hosts()[-1]
+        task_name = os.environ["AVALON_TASK"]
+        family = self.main_family_from_instance(instance)
+
+        self.log.info("Host: \"{}\"".format(host_name))
+        self.log.info("Task: \"{}\"".format(task_name))
+        self.log.info("Family: \"{}\"".format(family))
+
+        profile = self.find_matching_profile(
+            host_name, task_name, family
+        )
+        if not profile:
+            self.log.info((
+                "Skipped instance. None of profiles in presets are for"
+                " Host: \"{}\" | Family: \"{}\" | Task \"{}\""
+            ).format(host_name, family, task_name))
+            return
+
+        self.log.debug("Matching profile: \"{}\"".format(json.dumps(profile)))
+
+        instance_families = self.families_from_instance(instance)
+        _profile_outputs = self.filter_outputs_by_families(
+            profile, instance_families
+        )
+        if not _profile_outputs:
+            self.log.warning((
+                "Skipped instance. All output definitions from selected"
+                " profile does not match to instance families. \"{}\""
+            ).format(str(instance_families)))
+            return
+
+        # Store `filename_suffix` to save arguments
+        profile_outputs = []
+        for filename_suffix, definition in _profile_outputs.items():
+            definition["filename_suffix"] = filename_suffix
```

Comment:
I thought that camelCase is used in instance or context data to be able find keys across plugins...

Reply:
As far as I know we are using for example `frame_start` as variable name but in `instance.data["frameStart"]` is used. There is only exception in burn-ins and context.data because of use in the Anatomy - that is something we could discuss - consistency is desirable.

Reply:
> I thought that camelCase is used in instance or context data to be able find keys across plugins...

Which for me means to see in which plugins are `instance.data` or `context.data` keys used. These keys are not used across plugins, these are temporary data for this plugin. They're not stored in `instance.data` or `context.data`. I think it's not wise to use them with camel case to not mess search of keys used in `instance.data` or `context.data`.